runtime/src/main/java/org/corfudb/runtime/object/transactions/OptimisticTransactionalContext.java

```diff
@@ -239,9 +244,19 @@ public long getConflictSetAndCommit(ConflictSetInfo conflictSet) {
             return NOWRITE_ADDRESS;
         }
 
-        // Write to the transaction stream if transaction logging is enabled
-        Set<UUID> affectedStreamsIds = new HashSet<>(getWriteSetInfo().getWriteSet().getEntryMap().keySet());
+        Set<UUID> affectedStreamsIds = new HashSet<>(getWriteSetInfo()
+            .getWriteSet().getEntryMap().keySet());
 
+        Set<UUID> streamTags = new HashSet<>();
+        affectedStreamsIds.forEach( s ->
+        {
+            if (streamToProxyMap.containsKey(s)) {
+                streamTags.addAll(streamToProxyMap.get(s).getStreamTags());
+            }
+        });
+        affectedStreamsIds.addAll(streamTags);
+
+        // Write to the transaction stream if transaction logging is enabled
         if (this.transaction.isLoggingEnabled()) {
             affectedStreamsIds.add(TRANSACTION_STREAM_ID);
         }
```

Comment:
You can remove this TRANSACTION_STREAM_ID and use the new method to add a transaction stream